### PR TITLE
All claims transform refactor

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
First, I'd like to start off by saying sorry for the huge diff. I didn't actually change any of the logic in here. I just moved things around to make it easier to understand and modify.

The submit transformer was getting pretty gnarly, so I wanted to clean it up. We've had some troubles merging in new stuff that this should help alleviate.

### Things this PR does
1) Moves all the submit transformer functions to their own file: `submit-transformer.js`
   - The idea here was that it'd clean up the `utils.jsx` file a bit
2) Moves all the associated tests to a similarly-named test file
3) Refactors the `transform` function to:
   a. Define all the transformations as separate functions (within the `transform` function)
   b. Use all those transformation functions in a well-defined sequence

## Testing done
- I made sure all the existing unit tests pass
- I added in a temporary unit test that ensures the output of the transform is exactly what it was before
  - I did this by
    1) Making a little unit test (`test.unit.spec.js`) that mirrored the schema tests (minus validation) and logged the output of the transform functions
    2) Ran that and copied the results into corresponding json files (`maximal-test.json` then has a `maximal-test.json` in the `refactor-test/` directory, for example)
    3) Modified that unit test to make sure the result from the transform function is exactly the same as it was before

This is only a temporary test. I've included it in the PR for you to run and ensure that it works as expected. Feel free to modify the test json files with the output you get using the `transform` function from another branch.

Once I delete the `refactor-test/` directory after the PR has been reviewed, that'll slim this down by 350+ LoC, bringing the diff to something more like `+510 -450`.

## Screenshots
N/A

## Acceptance criteria
- [x] The output of the `transform` function is exactly what it was before the refactor
- [x] The function is cleaner, easier to understand, and easier to modify

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by cvalarida** [comment]
@dmccurdybah I added you as a reviewer for visibility. I wasn't sure who else to reach out to on your team, so feel free to notify anybody else who should know about this change.

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        key !== 'other' &&
+        key !== 'otherExplanation' &&
+        PTSD_CHANGE_LABELS[key],
+    )
+    .map(key => PTSD_CHANGE_LABELS[key]);
+}
+
+export function transform(formConfig, form) {
+  // Define the transformations
+  const filterSelectedRatedDisabilities = formData =>
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/submit-transformer.js]: This isn't a thing anymore as of my merged PR, but you'll probably find that out when resolving merge conflicts

**Review by markgreenburg — COMMENTED** [review]
Looks good! Hesitate to approve just yet until I can take a look at this with latest master merged in - let me know when ready and I'll give it a quick re-review.

File: src/applications/disability-benefits/all-claims/tests/schema/refactor-test/minimal-test.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "ratedDisabilities": [
+      {
+        "name": "First Condition",
+        "ratedDisabilityId": "0",
+        "ratingDecisionId": "63655",
+        "diagnosticCode": 5238,
+        "decisionCode": "SVCCONNCTED",
+        "decisionText": "Service Connected",
+        "ratingPercentage": 100,
+        "disabilityActionType": "INCREASE"
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/tests/schema/refactor-test/minimal-test.json]: This should be `NONE` and I think there should be a `view:selected` as well?

**Review by markgreenburg — APPROVED** [review]
Looks good; this is gonna be great!

**Pull request merged**